### DISCUSSION TITLE
How to keep the dom structure of nested html

### DISCUSSION THREAD

**Discussion by wkl007** [ueberdosis/tiptap#2270] [General Questions] [answered]
Hi everyone, I just started to learn TipTap. I have a problem, how to keep the dom structure of nested html . This is the [test url](https://codesandbox.io/s/pedantic-snow-tii4n?file=/src/views/HomeView.vue:472-689).

I create a SpanMark

```javascript
const SpanMark = Mark.create({
  name: "SpanMark",
  priority: 110,
  parseHTML() {
    return [
      {
        tag: "span",
      },
    ];
  },
  renderHTML() {
    return ["span", mergeAttributes({}), 0];
  },
});
```

I set the TipTap content:

```html
<span>123<span>456</span></span>
```

Tiptap analysis result is:

```html
<span>123456</span>
```

I want to get the result:

```html
<span>123<span>456</span></span>
```

Please help me , Thank you so much !!!

**Comment by philippkuehn** [comment, marked as answer]
Marks can’t be nested. You probably want to register an inline node for it.

**Reply by wkl007** [reply]
@philippkuehn Sorry I don't how to register an inline node for it. Can you provide a demo code?

**Reply by philippkuehn** [reply]
```js
Node.create({
  // ...
  inline: true,
})
```

**Reply by wkl007** [reply]
Thank you so much

**Reply by CarlMenke** [reply]
hey @wkl007 any chance you could share your final solution?

**Reply by fpernin-bkg** [reply]
Yes i would like to see the result and script
